framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -131,43 +121,64 @@ NearestNodeLocator::findNodes()
       }
     }
 
-    // don't need the BB anymore
-    delete my_inflated_box;
-
     const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map =
         _mesh.nodeToElemMap();
 
+    if (_ghost_elements)
+    {
```

Comment:
I tried using `FEProblemBase::addGhostedBoundary()` instead of `FEProblemBase::addGhostedElem(dof_id)` but it was causing a segmentation fault. There is a `_ghosted_elements` member variable in FEProblemBase which does not get filled if addGhostedBoundary is used.